doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -184,18 +185,246 @@ if transceiver is not present:
+                  Note over OA: Disable admin status<br>setPortSerdesAttribute
+                  OA ->> APPL_DB: PORT_TABLE:<lport>.MEDIA_SETTINGS_SYNC_STATUS = MEDIA_SETTINGS_DONE
+                  Note over OA: initHostTxReadyState
+                end
+            end
+        else
+            Note over SfpStateUpdateTask: retry_eeprom_set.add(lport)
+        end
+    end
+    Note over SfpStateUpdateTask: _init_port_sfp_status_tbl<br>Subscribe to CONFIG_DB
+    loop Wait for stop_event
+        SfpStateUpdateTask ->> SfpStateUpdateTask: Handle config change event<br>retry_eeprom_reading()<br>_wrapper_get_transceiver_change_event
+    end
+```
+
+## CMIS State machine with CMIS_STATE_MEDIA_SETTINGS_WAIT state
+
+The below state machine is a high level flow and doesn't capture details for states other than CMIS_STATE_MEDIA_SETTINGS_WAIT
+
+```mermaid
+stateDiagram
+    [*] --> CMIS_STATE_INSERTED
+    state if_state <<choice>>
+    state if_state2 <<choice>>
+    CMIS_STATE_INSERTED --> if_state
+    if_state --> CMIS_STATE_READY : if host_tx_ready != True or<br>admin_status != up<br> Action - disable TX
+    if_state --> if_state2 : if host_tx_ready == True and<br>admin_status == up
+    if_state2 --> CMIS_STATE_DP_DEINIT : if PORT_TABLE.port.CMIS_REINIT_REQUIRED == true or<br>is_cmis_application_update_required
+    if_state2 --> CMIS_STATE_MEDIA_SETTINGS_WAIT : if is_media_settings_supported and<br>MEDIA_SETTINGS_SYNC_STATUS != MEDIA_SETTINGS_DONE
+    note left of CMIS_STATE_READY : PORT_TABLE.port.CMIS_REINIT_REQUIRED = false
+    if_state2 --> CMIS_STATE_FAILED : if appl < 1 or <br>host_lanes_mask <= 0 or <br>media_lanes_mask <= 0
+    note left of CMIS_STATE_FAILED : PORT_TABLE.port.CMIS_REINIT_REQUIRED = false
+
+    CMIS_STATE_MEDIA_SETTINGS_WAIT --> CMIS_STATE_DP_DEINIT : if PORT_TABLE&ltport&gt.MEDIA_SETTINGS_SYNC_STATUS == MEDIA_SETTINGS_DONE
+    CMIS_STATE_MEDIA_SETTINGS_WAIT --> CMIS_STATE_INSERTED : Through force_cmis_reinit upon reaching timeout
+    note right of CMIS_STATE_MEDIA_SETTINGS_WAIT
+        Checks if PORT_TABLE&ltport&gt.MEDIA_SETTINGS_SYNC_STATUS == MEDIA_SETTINGS_DONE
+        After 5s timeout, force_cmis_reinit will be called
+    end note
+
+    CMIS_STATE_DP_DEINIT --> CMIS_STATE_AP_CONF
+    CMIS_STATE_AP_CONF --> CMIS_STATE_DP_INIT
+    CMIS_STATE_DP_INIT --> CMIS_STATE_DP_TXON
+    CMIS_STATE_DP_TXON --> CMIS_STATE_DP_ACTIVATE
+    CMIS_STATE_DP_ACTIVATE --> CMIS_STATE_READY
+```
+
+## Transceiver OIR handling
+
+```mermaid
+sequenceDiagram
+    participant STATE_DB
+    participant OA
+    participant APPL_DB
+    participant CmisManagerTask
+    participant SfpStateUpdateTask
+
+    SfpStateUpdateTask ->> SfpStateUpdateTask : event = SFP_STATUS_REMOVED
+    SfpStateUpdateTask -x STATE_DB : Delete TRANSCEIVER_INFO table for the port
+    par         CmisManagerTask, SfpStateUpdateTask
+        CmisManagerTask ->> CmisManagerTask : Transition CMIS SM to CMIS_STATE_REMOVED
```

Comment:
Should we set CMIS_REINIT_REQUIRED to True upon module removal?

Reply:
I am not currently changing the value of CMIS_REINIT_REQUIRED after setting it as False since the purpose of this flag is to drive CMIS re-initialization and not CMIS initialization.
The existing flow in the CMIS SM already ensures that CMIS initialization will be performed after module insertion.

Let me know your thoughts on this.